### PR TITLE
Сreated profile create and profile get workflows

### PR DESCRIPTION
**Pull request by Zhar7un**
Add profile entity, command, query, database provider

Swagger UI with the changes: 
image

### CODE REVIEW

File: Uanion/Uanion.Application/Features/Profile/Commands/CreateProfile/CreateProfileCommandValidator.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace Uanion.Application.Features.Profile.Commands.CreateProfile;
+
+public class CreateProfileCommandValidator : AbstractValidator<CreateProfileCommand>
+{
+    public CreateProfileCommandValidator()
+    {
+        RuleFor(p => p.UserId)
+            .NotNull();
+        //todo: check that this is the ID of an existing User
```
Comment (5x1fF1xgx1xti0n) [inline Uanion/Uanion.Application/Features/Profile/Commands/CreateProfile/CreateProfileCommandValidator.cs]: Please add private method below that uses user repository to check if user exists. Inject the user repository into CreateProfileCommandHandler and pass it here.

File: Uanion/Uanion.Application/Features/Profile/Queries/GetProfile/GetProfileQuery.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+using MediatR;
+
+namespace Uanion.Application.Features.Profile.Queries.GetProfile;
+
+public class GetProfileQuery : IRequest<ProfileViewModel>
+{
+    public Guid ProfileId { get; set; }
+}
```
Comment (5x1fF1xgx1xti0n) [inline Uanion/Uanion.Application/Features/Profile/Queries/GetProfile/GetProfileQuery.cs]: Add empty line in the end of file, please

File: Uanion/Uanion.Application/Features/Profile/Commands/CreateProfile/CreateProfileCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    public async Task<Guid> Handle(CreateProfileCommand request, CancellationToken cancellationToken)
+    {
+        var validatorResult = await _validator.ValidateAsync(request);
+
+        if (validatorResult.Errors.Count > 0)
+        {
+            throw new Exceptions.ValidationException(validatorResult);
+        }
+
+        var profile = _mapper.Map<Domain.Entities.Profile>(request);
```
Comment (5x1fF1xgx1xti0n) [inline Uanion/Uanion.Application/Features/Profile/Commands/CreateProfile/CreateProfileCommandHandler.cs:29]: That won't work. Please, register mapping profile for Profile entity. Look Uanion.Application -> Profiles -> UserMappingProfile

File: Uanion/Uanion.Application/Features/Profile/Queries/GetProfile/GetProfileQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    public GetProfileQueryHandler(IProfileRepository profileRepository, IMapper mapper)
+    {
+        _profileRepository = profileRepository;
+        _mapper = mapper;
+    }
+
+    public async Task<ProfileViewModel> Handle(GetProfileQuery request, CancellationToken cancellationToken)
+    {
+        var profile = await _profileRepository.GetByIdAsync(request.ProfileId);
+        var profileVm = _mapper.Map<ProfileViewModel>(profile);
```
Comment (5x1fF1xgx1xti0n) [inline Uanion/Uanion.Application/Features/Profile/Queries/GetProfile/GetProfileQueryHandler.cs:21]: That won't work. Please, register mapping profile for Profile entity. Look Uanion.Application -> Profiles -> UserMappingProfile

File: Uanion/Uanion.Persistence/Configurations/ProfileConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    public void Configure(EntityTypeBuilder<Profile> builder)
+    {
+        builder
+            .ToTable("Profile")
+            .HasKey(x => x.ProfileId);
+
+        builder
+            .Property(x => x.UserId)
+            .IsRequired();
+        //??
```
Comment (5x1fF1xgx1xti0n) [inline Uanion/Uanion.Persistence/Configurations/ProfileConfiguration.cs]: Remove the comment, please. These configurations are okay for now

**Review by 5x1fF1xgx1xti0n — CHANGES_REQUESTED** [review]
Some changes are missing:
1. As mentioned in comments below you have to add ProfileMappingProfile
2. Add ProfileController to API, ensure methods from the controller are displayed in SwaggerUI
3. Register ProfileRepository to DI as it's done with UserRepository (look PersistenceServiceRegistration class)
4. Add Profile DbSet to UanionDbContext
Thank you!

**Comment by Zhar7un** [comment]
@5x1fF1xgx1xti0n i found warning in handlers it asks to pass `cancellationToken` to `ValidateAsync` method call. I did it for profile handler. Perhaps we need to fix this in the user handler as well.

File: Uanion/Uanion.Application/Features/Profile/Commands/CreateProfile/CreateProfileCommandValidator.cs (hunk 1 above)
Comment (Zhar7un) [reply Uanion/Uanion.Application/Features/Profile/Commands/CreateProfile/CreateProfileCommandValidator.cs]: Done.

File: Uanion/Uanion.Application/Features/Profile/Queries/GetProfile/GetProfileQuery.cs (hunk 2 above)
Comment (Zhar7un) [reply Uanion/Uanion.Application/Features/Profile/Queries/GetProfile/GetProfileQuery.cs]: Done.

File: Uanion/Uanion.Application/Features/Profile/Queries/GetProfile/GetProfileQueryHandler.cs (hunk 4 above)
Comment (Zhar7un) [reply Uanion/Uanion.Application/Features/Profile/Queries/GetProfile/GetProfileQueryHandler.cs:21]: Done.

File: Uanion/Uanion.Application/Features/Profile/Commands/CreateProfile/CreateProfileCommandHandler.cs (hunk 3 above)
Comment (Zhar7un) [reply Uanion/Uanion.Application/Features/Profile/Commands/CreateProfile/CreateProfileCommandHandler.cs:29]: Done.

File: Uanion/Uanion.Persistence/Configurations/ProfileConfiguration.cs (hunk 5 above)
Comment (Zhar7un) [reply Uanion/Uanion.Persistence/Configurations/ProfileConfiguration.cs]: Done.

**Review by 5x1fF1xgx1xti0n — APPROVED** [review]
Lgtm, thank you!

**Pull request merged**